src/main/java/sc/fiji/hdf5/HDF5VirtualStack.java

```diff
@@ -0,0 +1,100 @@
+import ij.process.ByteProcessor;
+import ij.process.FloatProcessor;
+import ij.process.ImageProcessor;
+import ij.process.ShortProcessor;
+import org.apache.commons.lang.NotImplementedException;
+
+import java.util.Arrays;
+
+public class HDF5VirtualStack extends VirtualStack
+{
+  protected IHDF5Reader reader;
+  protected String dsetName;
+  protected int nRows, nCols, nLevels;
+  protected String typeText;
+  protected boolean isRGB;
+
+  public HDF5VirtualStack(IHDF5Reader reader, String dsetName, int nRows, int nCols, int nLevels, boolean isRGB, String typeText)
+  {
+    this.reader = reader;
+    this.dsetName = dsetName;
+    this.nRows = nRows;
+    this.nCols = nCols;
+    this.nLevels = nLevels;
+    this.typeText = typeText;
+    this.isRGB = isRGB;
+  }
+
+  @Override
+  public int getSize()
+  {
+    return this.nLevels;
+  }
+
+
+  @Override
+  public ImageProcessor getProcessor(int n)
+  {
+    n = n - 1; // zero index
+    ImageProcessor ip;
+    if (typeText.equals("uint8") && !isRGB)
+    {
+      MDByteArray rawdata = reader.uint8().readMDArrayBlock(dsetName, new int[]{1, nRows, nCols}, new long[]{n, 0, 0});
+      ip = new ByteProcessor(nCols, nRows);
+      System.arraycopy(rawdata.getAsFlatArray(), 0,
+          (byte[]) ip.getPixels(), 0,
+          nRows * nCols);
+    } else if (typeText.equals("uint8"))
+    {  // RGB data
+      throw new NotImplementedException("RGB 8 bit pixel images not supported in virtual stacks (yet)"); // I don't have a test file to check behavior on
+
+    } else if (typeText.equals("uint16"))
+    {
+      MDShortArray rawdata = reader.uint16().readMDArrayBlock(dsetName, new int[]{1, nRows, nCols}, new long[]{n, 0, 0});
+      ip = new ShortProcessor(nCols, nRows);
+      System.arraycopy(rawdata.getAsFlatArray(), 0,
+          (short[]) ip.getPixels(), 0,
+          nRows * nCols);
+    } else if (typeText.equals("int16"))
+    {
+      MDShortArray rawdata = reader.int16().readMDArrayBlock(dsetName, new int[]{1, nRows, nCols}, new long[]{n, 0, 0});
+      ip = new ShortProcessor(nCols, nRows);
+      System.arraycopy(rawdata.getAsFlatArray(), 0,
+          (short[]) ip.getPixels(), 0,
+          nRows * nCols);
+    } else if (typeText.equals("float32") || typeText.equals("float64"))
+    {
+      MDFloatArray rawdata = reader.float32().readMDArrayBlock(dsetName, new int[]{1, nRows, nCols}, new long[]{n, 0, 0});
+      ip = new FloatProcessor(nCols, nRows);
+      System.arraycopy(rawdata.getAsFlatArray(), 0,
+          (float[]) ip.getPixels(), 0,
+          nRows * nCols);
+    } else if (typeText.equals("int32"))
+    {
+      int[] rawdata_ints = reader.int32().readMDArrayBlock(dsetName, new int[]{1, nRows, nCols}, new long[]{n, 0, 0}).getAsFlatArray();
+      float[] rawdata = new float[rawdata_ints.length];
+      for (int i = 0; i < rawdata_ints.length; i++)
+      {
+        rawdata[i] = (float) rawdata_ints[i];
+      }
+      ip = new FloatProcessor(nCols, nRows);
```

Comment:
I think we want `IntProcessor` here.

https://github.com/imagej/ImageJ/blob/84e9e3b0c872b5c38121d0e77b6fbd4cfa4a3b9b/ij/process/IntProcessor.java